### PR TITLE
using config.json as a contracts config file

### PR DESCRIPTION
**Pull request by alexsdsr**

### CODE REVIEW

File: examples/migration/src/ledger.rs [hunk 1]
```diff
@@ -222,34 +222,60 @@ struct BesuContractConfig {
...
+        cur_dir.push("../../"); // project root directory
+        cur_dir.push(contract_path);
+        fs::canonicalize(&cur_dir)
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .to_string()
+    }
+
+    fn contracts() -> Vec<ContractConfig> {
```
Comment (Artemkaaas) [inline examples/migration/src/ledger.rs:240]: I think it makes sense to add explicit structure definition for the config file.
The function of reading the config will be much easier.

```
#[derive(Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
struct BesuContracts {
    did_registry: BesuContractConfig,
    schema_registry: BesuContractConfig,
    credential_definition_registry: BesuContractConfig,
    role_control: BesuContractConfig,
    legacy_mapping_registry: BesuContractConfig,
    .......
}
```

File: smart_contracts/demos/utils/actor.ts [hunk 2]
```diff
@@ -50,4 +54,19 @@ export class Actor {
   public get didDocument() {
     return this.account.didDocument
   }
+
+  private readContractsAddresses(): { [key: string]: string } {
```
Comment (Artemkaaas) [inline smart_contracts/demos/utils/actor.ts:58]: Let's just read contracts config without additional mapping

File: smart_contracts/scripts/genesis/config.ts [hunk 3]
```diff
@@ -24,130 +27,209 @@ export interface Config {
...
-  credentialDefinitionRegistry: '0x0000000000000000000000000000000000004444',
-  schemas: '0x0000000000000000000000000000000000005555',
-  roles: '0x0000000000000000000000000000000000006666',
-  validators: '0x0000000000000000000000000000000000007777',
-  accountControl: '0x0000000000000000000000000000000000008888',
-  upgradeControl: '0x0000000000000000000000000000000000009999',
-  legacyMappingRegistry: '0x0000000000000000000000000000000000019999',
+const contractsAddresses: { [key: string]: string } = {}
+
+export function readContractsConfig() {
```
Comment (Artemkaaas) [inline smart_contracts/scripts/genesis/config.ts:32]: move to helpers

File: smart_contracts/scripts/genesis/config.ts [hunk 4]
```diff
@@ -24,130 +27,209 @@ export interface Config {
...
+  }
+
+  if (contractsAddresses[contractName] === undefined) {
+    throw new Error(`contract '${contractName}' not found`)
+  }
+
+  return contractsAddresses[contractName]
 }
 
 export const config: Config = {
```
Comment (Artemkaaas) [inline smart_contracts/scripts/genesis/config.ts:56]: let's avoid using of global variable `contractsAddresses` and convert `config` into function reading config file and preparing data like now.  
Build config in the main. 
Pass sub-config into corresponding contract function.

File: vdr/src/client/client.rs [hunk 5]
```diff
@@ -275,39 +249,84 @@ pub mod test {
...
-            ContractConfig {
-                address: CRED_DEF_REGISTRY_ADDRESS.to_string(),
-                spec_path: Some(build_contract_path(CRED_DEF_REGISTRY_SPEC_PATH)),
-                spec: None,
-            },
-            ContractConfig {
-                address: VALIDATOR_CONTROL_ADDRESS.to_string(),
-                spec_path: Some(build_contract_path(VALIDATOR_CONTROL_PATH)),
-                spec: None,
+    fn get_contract_conf(contract_name: &str) -> Option<Value> {
```
Comment (Artemkaaas) [inline vdr/src/client/client.rs:252]: please add contract type definition for simplicity.

File: vdr/src/client/client.rs [hunk 6]
```diff
@@ -370,28 +389,28 @@ pub mod test {
 
         #[rstest]
```
Comment (Artemkaaas) [inline vdr/src/client/client.rs:390]: lets define lazy static variable holding read contracts config.

**Review by Artemkaaas — CHANGES_REQUESTED** [review]

File: vdr/src/client/client.rs [hunk 7]
```diff
@@ -240,23 +231,6 @@ pub mod test {
...
-        Lazy::new(|| Address::from("0x0000000000000000000000000000000000007777"));
-
-    pub static ROLE_CONTROL_ADDRESS: Lazy<Address> =
-        Lazy::new(|| Address::from("0x0000000000000000000000000000000000006666"));
-
-    pub static ETHR_DID_REGISTRY_ADDRESS: Lazy<Address> =
-        Lazy::new(|| Address::from("0x0000000000000000000000000000000000003333"));
-
-    pub static LEGACY_MAPPING_REGISTRY_ADDRESS: Lazy<Address> =
-        Lazy::new(|| Address::from("0x0000000000000000000000000000000000019999"));
```
Comment (Artemkaaas) [inline vdr/src/client/client.rs:259]: pub static CONTRACTS_CONFIG: Lazy<BesuConfig> =
        Lazy::new(|| read_config().uwnrap());
        

address: CONTRACTS_CONFIG.validator_control.address.to_string()

Comment (Artemkaaas) [reply vdr/src/client/client.rs:259]: #[serde(deny_unkonwn_fields)]

File: config.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
+{
+    "contracts": {
+        "did_validator": {
```
[the remark this answers is not in the document]
Comment (alexsdsr) [reply config.json:3]: done

File: examples/migration/src/ledger.rs (hunk 1 above)
Comment (alexsdsr) [reply examples/migration/src/ledger.rs:240]: done

File: smart_contracts/demos/utils/actor.ts (hunk 2 above)
Comment (alexsdsr) [reply smart_contracts/demos/utils/actor.ts:58]: done

File: vdr/src/client/client.rs (hunk 5 above)
Comment (alexsdsr) [reply vdr/src/client/client.rs:252]: done

File: vdr/src/client/client.rs (hunk 6 above)
Comment (alexsdsr) [reply vdr/src/client/client.rs:390]: done

**Review by Artemkaaas — APPROVED** [review]

**Pull request merged**